internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -0,0 +1,97 @@
+package osbuildexecutor
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"os/exec"
+
+	"github.com/osbuild/images/pkg/osbuild"
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/osbuild-composer/internal/cloud/awscloud"
+)
+
+type awsEC2Executor struct {
+	iamProfile string
+}
+
+func (ec2e *awsEC2Executor) RunOSBuild(manifest []byte, store, outputDirectory string, exports, exportPaths, checkpoints,
+	extraEnv []string, result bool, errorWriter io.Writer) (*osbuild.Result, error) {
+	region, err := awscloud.RegionFromInstanceMetadata()
+	if err != nil {
+		return nil, err
+	}
+
+	aws, err := awscloud.NewDefault(region)
+	if err != nil {
+		return nil, err
+	}
+
+	si, err := aws.RunSecureInstance(ec2e.iamProfile)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := aws.TerminateSecureInstance(si)
+		if err != nil {
+			logrus.Errorf("Error terminating secure instance: %v", err)
+		}
+	}()
+
+	logrus.Info("Spinning up jobsite manager")
+	args := []string{
+		"--builder-host",
+		*si.Instance.PrivateIpAddress,
+	}
+
+	for _, exp := range exports {
+		args = append(args, "--export", exp)
+	}
+	for _, exp := range exportPaths {
+		args = append(args, "--export-file", exp)
+	}
+	for _, env := range extraEnv {
+		args = append(args, "--environment", env)
+	}
+	args = append(args, "--output", outputDirectory)
+
+	cmd := exec.Command(
+		"/usr/libexec/osbuild-composer/osbuild-jobsite-manager",
+		args...,
+	)
+
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	err = cmd.Start()
```

Comment:
A bit of a meta-question - given that osbild-jobsite-manager is part of this repo and written in go, why is it not used directly from go code, why the extra indirection via a os/exec and the json? is it for robustness, i.e. a crashing osbuild-jobsite-manager would not affect the service? or is there a different/deeper reason?

Reply:
It's a layer of separation here; the worker might/will spawn multiple managers for multiple builders.